collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java

```diff
@@ -322,7 +323,18 @@ public void denied() {
                 // otherwise, do the normal aggregate/other thing.
                 Intent i = new Intent(this, InstanceUploaderActivity.class);
                 i.putExtra(FormEntryActivity.KEY_INSTANCES, instanceIds);
-                startActivityForResult(i, INSTANCE_UPLOADER);
+                // Not required but without this permission a Device ID attached to a request will be empty.
+                requestReadPhoneStatePermission(this, new PermissionListener() {
```

Comment:
@lognaturel what is the expected behavior here? Now we are asking for the permission but allow to continue in both cases (granted/denied) is that ok? If denied an empty value is sent instead of DeviceId.